modules/igv/1.0/igv.smk

```diff
@@ -0,0 +1,824 @@
+        target_reference = lambda w: config["lcr-modules"]["igv"]["options"]["liftover_regions"]["target_reference"][w.genome_build],
+        regions_type = lambda w: REGIONS_FORMAT[(w.tool_type).lower()],
+        regions_build = lambda w: (w.tool_build).replace("grch37","GRCh37").replace("hg38","GRCh38"),
+        target_build = lambda w: (w.genome_build).replace("grch37","GRCh37").replace("hg38","GRCh38")
+    conda:
+        CFG["conda_envs"]["crossmap"]
+    resources:
+        **CFG["resources"]["_igv_liftover_regions"]
+    log:
+        stdout = CFG["logs"]["inputs"] + "regions/liftover_regions_{tool_type}.{tool_build}To{genome_build}.stdout.log",
+        stderr = CFG["logs"]["inputs"] + "regions/liftover_regions_{tool_type}.{tool_build}To{genome_build}.stderr.log"
+    shell:
+        op.as_one_line("""
+        {input.liftover_script}
+        {input.regions} {params.regions_type} {params.regions_build}
+        {params.target_build} {output.regions} {params.chain_file}
+        {params.target_reference} > {log.stdout} 2> {log.stderr}
+        """)
+
+def _get_lifted_regions(wildcards):
+    CFG = config["lcr-modules"]["igv"]
+    return expand(
+        str(rules._igv_liftover_regions.output.regions),
+        tool_type = list(CFG["regions"]),
+        tool_build = ["grch37","hg38"],
+        allow_missing = True
+    )
+
+rule _igv_merge_lifted_regions:
+    input:
+        regions = _get_lifted_regions
+    output:
+        regions = CFG["dirs"]["inputs"] + "regions/regions.{genome_build}.txt"
+    run:
+        merged_df = pd.DataFrame()
+        for region in input.regions:
+            try:
+                df = pd.read_table(region, comment = "#", sep = "\t", header=None)
+                df.drop(df[df[0] == "chrom"].index, inplace = True)
+                merged_df = pd.concat([merged_df, df])
+            except:
+                print(f"Lifted regions file is empty: {region}")
+        merged_df = merged_df.drop_duplicates()
+        merged_df.to_csv(output.regions, sep="\t", na_rep="NA", index=False, header=False)
+
+# Filter MAF to lines containing positions of interest
+rule _igv_filter_maf:
+    input:
+        maf = str(rules._igv_reduce_maf_cols.output.maf),
+        regions = str(rules._igv_merge_lifted_regions.output.regions)
+    output:
+        maf = CFG["dirs"]["inputs"] + "maf/filtered_maf/{seq_type}--{genome_build}/{tumour_id}--{normal_sample_id}--{pair_status}.maf"
+    params:
+        regions_format = "bed",
+        metadata = CFG["runs"]
+    log:
+        stdout = CFG["logs"]["inputs"] + "filter_maf/{seq_type}--{genome_build}/{tumour_id}--{normal_sample_id}--{pair_status}.stdout.log",
+        stderr = CFG["logs"]["inputs"] + "filter_maf/{seq_type}--{genome_build}/{tumour_id}--{normal_sample_id}--{pair_status}.stderr.log"
+    wildcard_constraints:
+        seq_type = "[a-zA-Z]+"
+    script:
+        config["lcr-modules"]["igv"]["scripts"]["filter_script"]
+
+def _get_maf(wildcards):
```

Comment:
Curious why this does not conflict with the function defined above?
```
def get_maf(wildcards):
    unix_group = config["unix_group"]
    return expand(config["lcr-modules"]["igv"]["inputs"]["maf"], allow_missing=True, unix_group=unix_group)
```
Just because it is defined later the further call will use the "latest" definition?

Reply:
I removed the earlier function because I do not need it to fill in unix_group value (this can be done from the snakefile that is used to launch the run)